From 032b9b5b9e602f5e190c65ae8eb887c37c095bdb Mon Sep 17 00:00:00 2001
From: Tang Yun ping <typ@rock-chips.com>
Date: Mon, 11 Sep 2017 10:28:18 +0800
Subject: [PATCH] PM / devfreq: clk-ddr: fix bug of get lcdc type deadlock

When lcdc status changing and ddr frequency changing at the same time,
it may deadlock. Use list_for_each_entry to avoid this bug.

Change-Id: I461e705c19079ccb288e10bd4f616b0b4e5869b1
Signed-off-by: Tang Yun ping <typ@rock-chips.com>
---
 drivers/clk/rockchip/clk-ddr.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/clk/rockchip/clk-ddr.c b/drivers/clk/rockchip/clk-ddr.c
index d1d71c05a119..3bcf2bf758e7 100644
--- a/drivers/clk/rockchip/clk-ddr.c
+++ b/drivers/clk/rockchip/clk-ddr.c
@@ -53,14 +53,13 @@ static int rk_drm_get_lcdc_type(void)
 	if (drm) {
 		struct drm_connector *conn;
 
-		drm_modeset_lock(&drm->mode_config.connection_mutex, NULL);
-		drm_for_each_connector(conn, drm) {
+		list_for_each_entry(conn, &drm->mode_config.connector_list,
+				    head) {
 			if (conn->encoder) {
 				lcdc_type = conn->connector_type;
 				break;
 			}
 		}
-		drm_modeset_unlock(&drm->mode_config.connection_mutex);
 	}
 
 	switch (lcdc_type) {
-- 
2.35.3

